knowledge-base/grid-filterrow-bool-customization.md

```diff
@@ -0,0 +1,150 @@
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>Grid for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+
+## Description
+
+This KB article answers the following questions:
+
+* How to change default text `is true` and `is false` in `Grid FilterRow` dropdown list?
+* How to customize the `FilterRow` values?
+* How to change the default display names in `FilterRow` dropdown list?
+* How to filter a boolean `Grid` column values by `null`?
+* How to change the default values in FilterRow dropdown list to "**Yes**" and "**No**"?
+
+## Solution
+
+* Define a [Filter Row Template]({%slug grid-templates-filter%}#filter-row-template).
+* Create a `DropDownList` that includes the custom display values.
+* Create a `Button` that will replicate the `FilterRow` clear button.
+* Implement a method for manually filtering the `Grid` based on the dropdown list selection.
+
+>caption `Grid` with custom display values in the `FilterRow` dropdown list and manual filtering.
+
+````CSHTML
+@using Telerik.DataSource
+
+<TelerikGrid Data=@GridData FilterMode="GridFilterMode.FilterRow"
+             Pageable="true" Height="400px" Width="800px" PageSize="@GridData.Count()">
+    <GridColumns>
+        <GridColumn Field=@nameof(Employee.Name) />
+        <GridColumn Field=@nameof(Employee.IsOnLeave) Title="On Vacation">
+            <FilterCellTemplate>
+
+                <!-- Dropdown list for selecting filter options -->
+                <TelerikDropDownList Data="@FilterOptions" Value="@DropDownListValue"
+                                     ValueChanged="@((string newValue) => ApplyFilter(newValue, context))">
+                    <DropDownListSettings>
+                        <DropDownButtonPopupSettings Height="auto" />
+                    </DropDownListSettings>
+                </TelerikDropDownList>
+
+                <!-- Button to clear the filter -->
+                <TelerikButton ButtonType="ButtonType.Button"
+                               Class="k-clear-button-visible ml-2"
+                               Icon="@SvgIcon.FilterClear"
+                               Enabled="@(DropDownListValue != "All")"
+                               OnClick="@(async () =>
+                                          {
+                                              await context.ClearFilterAsync();
+                                              DropDownListValue = "All";
+                                          })">
+                </TelerikButton>
+            </FilterCellTemplate>
+        </GridColumn>
+    </GridColumns>
+</TelerikGrid>
+
+@code {
+    private List<Employee> GridData { get; set; } = new();
+
+    // Custom options for the dropdown filter
+    private List<string> FilterOptions { get; set; } = new() { "All", "Yes", "No", "No Info" };
+
+    // Selected value in the dropdown filter
+    private string DropDownListValue { get; set; } = "All";
```

Comment:
Using hard-coded strings in the app's business logic is not a very good practice. Imagine that the customer is working on localized app, which serves different languages.